documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,947 @@
+
+Public API for almost all actions and operations has changed significantly. For more information about the new features
+and specific code examples, refer to the [How to get started with AWS IoT Device SDK for Java v2](#how-to-get-started-with-aws-iot-device-sdk-for-java-v2)
+section of this guide.
+
+## How to get started with AWS IoT Device SDK for Java v2
+
+There're differences between the v1 SDK and the v2 SDK. This section describes the changes you need to apply to your project
+with the v1 SDK to start using the v2 SDK. For more information about MQTT5, visit [MQTT5 User Guide](https://github.com/aws/aws-iot-device-sdk-java-v2/blob/main/documents/MQTT5_Userguide.md#getting-started-with-mqtt5).
+
+### Package name change
+
+A noticeable change from the v1 SDK to the v2 SDK is the package name change. Package names begin with `software.amazon.awssdk`
+in the v2 SDK, whereas the v1 SDK uses `com.amazonaws`.
+
+These same names differentiate Maven artifacts from v1 to v2. Maven artifacts for the v2 SDK use the `software.amazon.awssdk`
+groupId, whereas the v1 SDK uses the `com.amazonaws` groupId.
+
+#### Example of adding the v1 SDK to Maven project
+
+```
+<dependency>
+  <groupId>com.amazonaws</groupId>
+  <artifactId>aws-iot-device-sdk-java</artifactId>
+  <version>1.3.9</version>
+</dependency>
+```
+
+#### Example of adding the v2 SDK to Maven project
+
+```
+<dependency>
+  <groupId>software.amazon.awssdk.iotdevicesdk</groupId>
+  <artifactId>aws-iot-device-sdk</artifactId>
+  <version>1.19.0</version>
+</dependency>
+```
+
+### MQTT protocol
+
+The v1 SDK uses an MQTT version 3.1.1 client under the hood.
+
+The v2 SDK provides MQTT version 3.1.1 and MQTT version 5.0 client implementations. This guide focuses on MQTT5 because
+this version is a significant improvement over MQTT3. For more information, see the [MQTT5 features](#mqtt5-features) section.
+
+### Client builder
+
+To access AWS IoT services, you must initialize an MQTT client.
+
+In the v1 SDK, the [AWSIotMqttClient](http://aws-iot-device-sdk-java-docs.s3-website-us-east-1.amazonaws.com/com/amazonaws/services/iot/client/AWSIotMqttClient.html)
+class represents an MQTT client. You instantiate the client directly by passing all the required parameters to the class
+constructor. It's possible to change the client settings after its creation using the `set*` methods like `setKeepAliveInterval`
+or `setMaxConnectionRetries`.
+
+In the v2 SDK, the [Mqtt5Client](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/Mqtt5Client.html)
+class represents an MQTT client, specifically for MQTT5 protocol. The v2 SDK provides an [MQTT5 client builder](https://aws.github.io/aws-iot-device-sdk-java-v2/software/amazon/awssdk/iot/AwsIotMqtt5ClientBuilder.html)
+designed to easily create common configuration types such as direct MQTT or WebSocket connections. After an MQTT5 client
+is built and finalized, the settings of the resulting MQTT5 client cannot be modified.
+
```

Comment:
This is where we might inject the connect/start info.

Reply:
I tried to mention start and stop here, but it looked out of place. I think, the corresponding sections are visible enough, so I decided to leave this section as is.